src/expertsystem/reaction/__init__.py

```diff
@@ -429,6 +453,184 @@ def _solve(
+    if not isinstance(initial_state, (list, tuple)):
+        initial_state = [initial_state]  # type: ignore
+    topology_builder = SimpleStateTransitionTopologyBuilder(
+        [
+            InteractionNode(
+                "NBodyScattering", len(initial_state), len(final_state)
+            )
+        ]
+    )
+    topology = topology_builder.build_graphs(
+        len(initial_state), len(final_state)
+    )[0]
+
+    # Step 2: initialize topologies with initial and final state
+    initialized_graphs = initialize_graph(
+        topology=topology,
+        particles=load_default_particles(),
+        initial_state=initial_state,
+        final_state=final_state,
+    )
+
+    # EdgeRules: (can be checked for each node individually)
+    pure_edge_rules: Set[GraphElementRule] = {
+        gellmann_nishijima,
+        isospin_validity,
+    }
+
+    # Step 4: verify edges with the edge rules first
+    edge_check_result = validate_fully_initialized_graph(
+        initialized_graphs[0],
+        rules_per_node={},
+        rules_per_edge={
+            edge_id: pure_edge_rules
+            for edge_id in initialized_graphs[0].get_initial_state_edges()
+            + initialized_graphs[0].get_final_state_edges()
+        },
+    )
+
+    if edge_check_result.violated_edge_rules:
+        raise ValueError(
+            f"Some edges violate"
+            f" {edge_check_result.violated_edge_rules.values()}"
+        )
+
+    # Those rules give the same results, independent on the node and spin props.
+    # Note they are also independent of the topology and hence their results
+    # are always correct.
+    edge_qn_conservation_rules: Set[Rule] = {
+        BaryonNumberConservation(),
+        BottomnessConservation(),
+        ChargeConservation(),
+        CharmConservation(),
+        ElectronLNConservation(),
+        MuonLNConservation(),
+        StrangenessConservation(),
+        TauLNConservation(),
+        isospin_conservation,
+    }
+    if len(initial_state) == 1:
+        edge_qn_conservation_rules.add(MassConservation(5))
+
+    violations: Set[Tuple[str, ...]] = {
+        (x,)
+        for x in _check_violations(
+            initialized_graphs[0], edge_qn_conservation_rules
+        )
+    }
+
+    # Step 5: Create combinations of graphs for magnitudes of S and L, but only
+    # if it is a two body reaction
+    ls_combinations = [
+        InteractionProperties(l_magnitude=l_mag, s_magnitude=s_mag)
+        for l_mag, s_mag in product([0, 1], [0, 0.5, 1, 1.5, 2])
+    ]
+    node_id = list(initialized_graphs[0].nodes)[0]
+    graphs = []
+    for ls_combi in ls_combinations:
+        for graph in initialized_graphs:
+            new_graph = copy(graph)
+            new_graph.node_props = {node_id: ls_combi}
+            graphs.append(new_graph)
+
+    # Step 6: Then verify each graph with the interaction rules. Process each
+    # rule separately and keep track of the on which graph each rule passes or
```

Comment:
~the on~

Reply:
44e8ea04bba93a7a35021d254c48a9afa1d045b3